napari/_vispy/canvas.py

```diff
@@ -70,26 +200,398 @@ def _set_theme_change(self, theme: str):
         )[0]
         self.bgcolor = self._last_theme_color
 
+    def _disconnect_theme(self) -> None:
+        self.viewer.events.theme.disconnect(self._on_theme_change)
+
     @property
-    def bgcolor(self):
-        SceneCanvas.bgcolor.fget(self)
+    def bgcolor(self) -> str:
+        """Background color of the vispy scene canvas as a hex string. The setter expects str
+        (any in vispy.color.get_color_names) or hex starting with # or a tuple | np.array ({3,4},)
+        with values between 0 and 1."""
+        return self._scene_canvas.bgcolor.hex
 
     @bgcolor.setter
-    def bgcolor(self, value):
-        _value = self._background_color_override or value
-        SceneCanvas.bgcolor.fset(self, _value)
+    def bgcolor(self, value: npt.ArrayLike) -> None:
+        self._scene_canvas.bgcolor = value
 
     @property
-    def central_widget(self):
+    def central_widget(self) -> Widget:
         """Overrides SceneCanvas.central_widget to make border_width=0"""
-        if self._central_widget is None:
-            self._central_widget = Widget(
-                size=self.size, parent=self.scene, border_width=0
+        if self._scene_canvas._central_widget is None:
+            self._scene_canvas._central_widget = Widget(
+                size=self.size,
+                parent=self._scene_canvas.scene,
+                border_width=0,
             )
-        return self._central_widget
+        return self._scene_canvas._central_widget
 
-    def _process_mouse_event(self, event):
-        """Ignore mouse wheel events which have modifiers."""
-        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+    @property
+    def size(self) -> Tuple[int, int]:
+        """Return canvas size as tuple (height, width) or accepts size as tuple (height, width)
+        and sets Vispy SceneCanvas size as (width, height)."""
```

Comment:
Maybe a bit confusing to point out this xy swap in the docstring, it's just an implementation detail, so I'd rather put it in a comment.

Reply:
The reason why I put it in the docstring is to make it explicit what kind of convention to use, given the confusion regarding what convention we are using on the napari side.